ledger/acctupdates.go

```diff
@@ -1008,7 +1008,20 @@ func (au *accountUpdates) commitRound(offset uint64, dbRound basics.Round, lookb
 	var catchpointLabel string
 	beforeUpdatingBalancesTime := time.Now()
 	var trieBalancesHash crypto.Digest
+
+	// the lockTaken variable would help us to ensure that we take the lock before commiting the transaction to the database.
+	// since the function within the "Atomic" migth be re-attempted, we need to perform the check at the begining, and after
+	// we exit from Atomic. Note that there is no concurrently issue within this function around testing the variable value.
```

Comment:
I guess more explanation needed - atomic function might be successful but re-attempted if commit inside atomic failed.

Reply:
updated.